### PR TITLE
ClientInvokeCallback tests should always reset ClientInvokeCallback

### PR DESCRIPTION
**Pull request by ReubenBond**
We are seeing failures related to this in functional test runs recently.
One of the test cases is not resetting ClientInvokeCallback.

This PR wraps each of those tests in try/finally blocks to reset `GrainClient.ClientInvokeCallback` and also resets it in the classes `Dispose()` method.

### CODE REVIEW

File: test/TesterInternal/General/RequestContextTest.cs [hunk 1]
```diff
@@ -741,58 +742,84 @@ public void Dispose()
...
+                Assert.Equal(1, callback.TotalCalls); // "Number of callbacks"
+
+                GrainClient.ClientInvokeCallback = null;
+                activityId = await grain.E2EActivityId();
+                Assert.Equal(Guid.Empty, activityId); // "E2EActivityId Call#2"
+                Assert.Equal(1, callback.TotalCalls); // "Number of callbacks - should be unchanged"
+            }
+            finally
+            {
+                GrainClient.ClientInvokeCallback = null;
```
Comment (ReubenBond) [inline test/TesterInternal/General/RequestContextTest.cs]: I wasn't certain if it was - I'll revert those bits.

File: Clean.cmd [hunk 2]
```diff
@@ -0,0 +1,5 @@
+@ECHO .
+@ECHO Cleaning %CD%
+@ECHO .
+FOR /F "tokens=*" %%G IN ('DIR /B /AD /S bin') DO RMDIR /S /Q "%%G"
+FOR /F "tokens=*" %%G IN ('DIR /B /AD /S obj') DO RMDIR /S /Q "%%G"
```
Comment (jdom) [inline Clean.cmd]: unrelated change?
